lib/src/block_parser.dart

```diff
@@ -136,13 +136,13 @@ class BlockParser {
   /// Gets whether or not the current line matches the given pattern.
   bool matches(RegExp regex) {
     if (isDone) return false;
-    return regex.firstMatch(current) != null;
+    return regex.hasMatch(current);
```

Comment:
Do you know why this is so much faster?

Reply:
Those might not affect performance, but they're simpler, and they _might_ affect performance, if it means no Match object allocations.

Reply:
Ahh, okay - the performance gain here is coming from the smaller regexp check you added